mne/io/neuralynx/neuralynx.py

```diff
@@ -98,31 +139,133 @@ def __init__(
         # the sample sizes of all segments
         n_segments = nlx_reader.header["nb_segment"][0]
         block_id = 0  # assumes there's only one block of recording
-        n_total_samples = sum(
-            nlx_reader.get_signal_size(block_id, segment)
-            for segment in range(n_segments)
+
+        # get segment start/stop times
+        start_times = np.array(
+            [nlx_reader.segment_t_start(block_id, i) for i in range(n_segments)]
+        )
+        stop_times = np.array(
+            [nlx_reader.segment_t_stop(block_id, i) for i in range(n_segments)]
         )
 
-        # construct an array of shape (n_total_samples,) indicating
-        # segment membership for each sample
-        sample2segment = np.concatenate(
-            [
-                np.full(shape=(nlx_reader.get_signal_size(block_id, i),), fill_value=i)
-                for i in range(n_segments)
+        # find discontinuous boundaries (of length n-1)
+        next_start_times = start_times[1::]
+        previous_stop_times = stop_times[:-1]
+        seg_diffs = next_start_times - previous_stop_times
+
+        # mark as discontinuous any two segments that have
+        # start/stop delta larger than sampling period (1/sampling_rate)
+        logger.info("Checking for temporal discontinuities in Neo data segments.")
+        delta = 1 / info["sfreq"]
+        gaps = seg_diffs > delta
+        has_gaps = gaps.any()
+
+        seg_gap_dict = {}
+
+        if has_gaps:
+            logger.info(
+                f"N = {gaps.sum()} discontinuous Neo segments detected "
+                + f"with delta > {delta} sec.\n"
+                + f"(max = {seg_diffs[gaps].max()} sec, min = {seg_diffs[gaps].min()})"
+                + "\nAnnotating gaps as BAD_ACQ_SKIP."
+            )
+
+            gap_starts = stop_times[:-1][gaps]  # gap starts at segment offset
+            gap_stops = start_times[1::][gaps]  # gap stops at segment onset
+
+            # (n_gaps,) array of ints giving number of samples per inferred gap
+            gap_n_samps = np.array(
+                [
+                    len(np.arange(on_spl, off_spl))
+                    for on_spl, off_spl in zip(
+                        gap_starts * info["sfreq"], gap_stops * info["sfreq"]
+                    )
```

Comment:
Use int arithmetic where possible rather than relying on `np.arange` over float with a step of `1.`, just figure out how many elements there should be
```suggestion
                    int(round(stop * info["sfreq"])) - int(round(start * info["sfreq"]))
                    for start, stop in zip(gap_starts, gap_stops)
```

Reply:
added in https://github.com/mne-tools/mne-python/pull/12279/commits/2c0861c03ecd79b1e65115440318929c4f3acdb3